src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js

```diff
@@ -63,224 +64,217 @@ const defaultProps = {
     preferredLocale: CONST.LOCALES.DEFAULT,
 };
 
-class BaseValidateCodeForm extends React.Component {
-    constructor(props) {
-        super(props);
-        this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
-        this.resendValidateCode = this.resendValidateCode.bind(this);
-        this.clearSignInData = this.clearSignInData.bind(this);
+function BaseValidateCodeForm(props) {
+    const [formError, setFormError] = useState({});
+    const [validateCode, setValidateCode] = useState(props.credentials.validateCode || '');
+    const [twoFactorAuthCode, setTwoFactorAuthCode] = useState('');
+    const [linkSent, setLinkSent] = useState(false);
 
-        this.state = {
-            formError: {},
-            validateCode: props.credentials.validateCode || '',
-            twoFactorAuthCode: '',
-            linkSent: false,
-        };
-    }
+    const wasVisible = usePrevious(props.isVisible);
+    const required2FA = usePrevious(props.account.requiresTwoFactorAuth);
+    const hadValidateCode = usePrevious(props.credentials.validateAndSubmitForm);
 
-    componentDidMount() {
-        if (!canFocusInputOnScreenFocus() || !this.inputValidateCode || !this.props.isVisible) {
+    const inputValidateCodeRef = useRef();
+    const input2FARef = useRef();
+
+    useEffect(() => {
+        if (!inputValidateCodeRef.current || wasVisible || !props.isVisible || !canFocusInputOnScreenFocus()) {
             return;
         }
-        this.inputValidateCode.focus();
-    }
+        inputValidateCodeRef.current.focus();
+    }, [inputValidateCodeRef, props.isVisible, wasVisible]);
 
-    componentDidUpdate(prevProps) {
-        if (!prevProps.isVisible && this.props.isVisible) {
-            this.inputValidateCode.focus();
-        }
-        if (prevProps.isVisible && !this.props.isVisible && this.state.validateCode) {
-            this.clearValidateCode();
+    useEffect(() => {
+        if (!inputValidateCodeRef.current) {
+            return;
         }
 
-        // Clear the code input if a new magic code was requested
-        if (this.props.isVisible && this.state.linkSent && this.props.account.message && this.state.validateCode) {
-            this.clearValidateCode();
+        // Clear the code input if magic code valid or a new magic code was requested
+        if ((wasVisible && !props.isVisible && validateCode) || (props.isVisible && linkSent && props.account.message && validateCode)) {
+            setValidateCode('');
+            inputValidateCodeRef.current.clear();
```

Comment:
Can you rewrite this in a new function as it is now (`clearValidateCode`). And the `clear` method was called after setting the state, can we guarantee that as well

Reply:
I'm not sure about that. Is there a problem if we use `useEffect`?

Reply:
Should use `useEffect` now to clear both inputs when they have empty strings.